ui/apps/platform/cypress/helpers/tableHelpers.ts

```diff
@@ -0,0 +1,59 @@
+export function getTableRowLinkByName(name: string) {
+    const exactName = new RegExp(`^${name}$`, 'g');
+    return cy
+        .get('a')
+        .contains(exactName)
+        .then(($el) => {
+            return cy.wrap($el);
+        });
+}
+
+export function getTableRowActionButtonByName(name: string) {
+    const exactName = new RegExp(`^${name}$`, 'g');
+    return cy
+        .get('a')
+        .contains(exactName)
+        .then(($el) => {
+            return cy.wrap($el).parent().siblings('td').find('button[aria-label="Kebab toggle"]');
+        });
+}
+
+export function editIntegration(name: string) {
+    cy.get(`tr:contains('${name}') td.pf-v5-c-table__action button`).click();
+    cy.get(
+        `tr:contains('${name}') td.pf-v5-c-table__action button:contains("Edit Integration")`
+    ).click();
+}
+
+export function queryTableHeader(headerName: string) {
+    return cy.get(`th`).contains(new RegExp(`^${headerName}$`));
+}
+
+export function queryTableSortHeader(headerName: string) {
+    return cy
+        .get(`th button:has('.pf-v5-c-table__sort-indicator')`)
+        .contains(new RegExp(`^${headerName}$`));
+}
+
+export function sortByTableHeader(headerName: string) {
+    return queryTableSortHeader(headerName).click();
+}
+
+export function assertOnEachRowForColumn(
+    columnDataLabel: string,
+    // eslint-disable-next-line no-unused-vars
+    assertion: (index: number, el: HTMLElement) => void
+) {
+    return cy
+        .get(`table:has(th:contains("${columnDataLabel}"))`)
+        .then(($el) => $el.find(`td[data-label="${columnDataLabel}"]`))
+        .then(($cells) => {
+            if ($cells.length === 0) {
+                cy.log(
+                    `No rows found for the column [${columnDataLabel}], assertion will pass applied to 0 rows`
+                );
+            }
+            $cells.each(assertion);
+            return Promise.resolve();
+        });
+}
```

Comment:
This function is the only new addition to this JS->TS file.

Since we cannot reliable assert on data in CI, this function provides an alternative:
1. Query table rows off the top level `table` via jQuery to avoid Cypress errors when no rows exist
2. Run an assertion against each visible row in the table, if available
3. If no rows are available, the assertion exits early without failure. This isn't ideal, but is equivalent to not testing table properties at all. At worst we are equal with no testing, at best we do get some verification of e2e functionality.

The idea here being that we test some invariant of the table, based on applied filters, etc, instead of testing specifics of data.

e.g. If a filter of `CVSS:>7` is applied, we can assert that every table row has a numeric text value >7 in the table under the "CVSS" column heading.

e.g. If a filter of `Severity:CRITICAL` is applied, we can assert every table row has a "Critical" badge with a numeric text value >0, and "Important", "Moderate", "Low" badges that are "hidden".

In each case if there is no data returned by the BE, this function will not cause an error in the test.